### ISSUE TITLE
When running Play from sbt when piping input/output, web-application shuts down immediately after starting

### ISSUE THREAD

**Issue by MikeEngUSA** [playframework/playframework#11086]
I am using _Play_ with _IntelliJ IDEA 2021.3.1 (Ultimate Edition)_, together with the _IntelliJ Play Framework_ plugin (v 213.5744.125). I am reporting this problem here, because I believe the issue lies with how the _Play_ web-server executes, rather than with how _IntelliJ_ runs the _Play_ application. The application is using _SBT_ v1.5.8 to build, with _Scala_ 2.13.7.

The problem is that the _Play_ web server immediately shuts down after starting (via either the _IntelliJ Play Framework_ plugin's _Play 2 App_ or  `sbt run` run configurations), while piping it's terminal input/output to _IntelliJ_. This did not happen with the previous Play version that I was using (v2.8.8). Furthermore, it's still possible to run `sbt run` from the command line, and have the web application start successfully. However, I could be wrong, and it's possible this is a latent issue with _IntelliJ IDEA_.

I suspect that this may be a symptom of the recent fix to #11032, and the changes made between release 2.8.8 and 2.8.11 that caused that issue to arise.

### Play Version
2.8.12

### API

Scala

### Operating System

Windows 10 Pro 64-bit (21H1)

### JDK

openjdk version "11.0.13" 2021-10-19 LTS
OpenJDK Runtime Environment Zulu11.52+13-CA (build 11.0.13+8-LTS)
OpenJDK 64-Bit Server VM Zulu11.52+13-CA (build 11.0.13+8-LTS, mixed mode)

### Expected Behavior

1. Start web-application from within _IntelliJ_ using _IntelliJ Play Framework_'s _Play 2 App_ run configuration, or a regular _IntelliJ_ _sbt run task_ run configuration.

### Actual Behavior

1. Web application starts, then immediately terminates:

Output from a "Play 2 App" run configuration:

```plain
"C:\Program Files\Zulu\jdk-11\bin\java.exe" -Dfile.encoding=UTF8 -Djline.terminal=none -Dsbt.global.base=C:\Users\Me\AppData\Local\Temp\sbt-global-pluginstub -Dsbt.log.noformat=true -XX:+CMSClassUnloadingEnabled -classpath C:\Users\Me\AppData\Roaming\JetBrains\IntelliJIdea2021.3\plugins\Scala\launcher\sbt-launch.jar xsbt.boot.Boot "project root" run
[info] welcome to sbt 1.5.8 (Azul Systems, Inc. Java 11.0.13)
[info] loading settings for project mywebapp-build from CoffeeScript.sbt,Play.sbt,Release.sbt ...
[info] loading project definition from C:\Users\Me\src\MyWebApp\project
[info] loading settings for project root from build.sbt,version.sbt ...
[info]   __              __
[info]   \ \     ____   / /____ _ __  __
[info]    \ \   / __ \ / // __ `// / / /
[info]    / /  / /_/ // // /_/ // /_/ /
[info]   /_/  / .___//_/ \__,_/ \__, /
[info]       /_/               /____/
[info] 
[info] Version 2.8.12 running Java 11.0.13
[info] 
[info] Play is run entirely by the community. If you want to keep using it please consider donating:
[info] https://www.playframework.com/sponsors
[info] 
[info]   __              __
[info]   \ \     ____   / /____ _ __  __
[info]    \ \   / __ \ / // __ `// / / /
[info]    / /  / /_/ // // /_/ // /_/ /
[info]   /_/  / .___//_/ \__,_/ \__, /
[info]       /_/               /____/
[info] 
[info] Version 2.8.12 running Java 11.0.13
[info] 
[info] Play is run entirely by the community. If you want to keep using it please consider donating:
[info] https://www.playframework.com/sponsors
[info] 

--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.AkkaHttpServer - Listening for HTTP on /127.0.0.1:9000

Server started, use Alt+D to stop

[info] p.a.i.l.c.CoordinatedShutdownSupport - Starting synchronous coordinated shutdown with ServerStoppedReason reason and 35000 milliseconds timeout
[info] p.c.s.AkkaHttpServer - Stopping Akka HTTP server...
[info] p.c.s.AkkaHttpServer - Terminating server binding for /127.0.0.1:9000
[info] p.c.s.AkkaHttpServer - Running provided shutdown stop hooks

[success] Total time: 3 s, completed Jan 6, 2022, 4:04:42 PM

Process finished with exit code 0
```

Output from a `sbt run` _sbt Task_ run configuration:
```plain
"C:\Program Files\Zulu\jdk-11\bin\java.exe" -XX:+CMSClassUnloadingEnabled "-javaagent:C:\Program Files\JetBrains\IDEAU\lib\idea_rt.jar=53264:C:\Program Files\JetBrains\IDEAU\bin" -Dfile.encoding=windows-1252 -classpath C:\Users\Me\AppData\Roaming\JetBrains\IntelliJIdea2021.3\plugins\Scala\launcher\sbt-launch.jar xsbt.boot.Boot run
[info] welcome to sbt 1.5.8 (Azul Systems, Inc. Java 11.0.13)
[info] loading global plugins from C:\Users\Me\.sbt\1.0\plugins
[info] loading settings for project mywebapp-build from CoffeeScript.sbt,Play.sbt,Release.sbt ...
[info] loading project definition from C:\Users\Me\src\MyWebApp\project
[info] loading settings for project root from build.sbt,version.sbt ...
[info]   __              __
[info]   \ \     ____   / /____ _ __  __
[info]    \ \   / __ \ / // __ `// / / /
[info]    / /  / /_/ // // /_/ // /_/ /
[info]   /_/  / .___//_/ \__,_/ \__, /
[info]       /_/               /____/
[info] 
[info] Version 2.8.12 running Java 11.0.13
[info] 
[info] Play is run entirely by the community. If you want to keep using it please consider donating:
[info] https://www.playframework.com/sponsors
[info] 

--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.AkkaHttpServer - Listening for HTTP on /127.0.0.1:9000

(Server started, use Enter to stop and go back to the console...)

[info] p.a.i.l.c.CoordinatedShutdownSupport - Starting synchronous coordinated shutdown with ServerStoppedReason reason and 35000 milliseconds timeout
[info] p.c.s.AkkaHttpServer - Stopping Akka HTTP server...
[info] p.c.s.AkkaHttpServer - Terminating server binding for /127.0.0.1:9000
[info] p.c.s.AkkaHttpServer - Running provided shutdown stop hooks

[success] Total time: 5 s, completed Jan 6, 2022, 4:07:38 PM

Process finished with exit code 0
```

**Comment by MikeEngUSA** [comment]
In fact, the same problem occurs if the web application is invoked (as `sbt run`) from a _Windows_ batch file in which output is redirected to a file, so this issue has nothing to do with _IntelliJ_.

Doing this used to work with _Play_ v2.8.8.

To reproduce, create a _Windows_ batch file, called, say, `test.bat` with the following contents:
```bat
cd C:\Path\To\Your\Web\App
call sbt run > test.log 2>&1
```
When you run this command, you'll see `test.log` contains the following:
```plain
[info] welcome to sbt 1.5.8 (Azul Systems, Inc. Java 11.0.11)
[info] loading settings for project mywebapp-build from CoffeeScript.sbt,Play.sbt,Release.sbt ...
[info] loading project definition from C:\Path\To\Your\WebApp\project
[info] loading settings for project root from build.sbt,version.sbt ...
[info]   __              __
[info]   \ \     ____   / /____ _ __  __
[info]    \ \   / __ \ / // __ `// / / /
[info]    / /  / /_/ // // /_/ // /_/ /
[info]   /_/  / .___//_/ \__,_/ \__, /
[info]       /_/               /____/
[info]
[info] Version 2.8.12 running Java 11.0.13
[info]
[info] Play is run entirely by the community. If you want to keep using it please consider donating:
[info] https://www.playframework.com/sponsors
[info]

--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.AkkaHttpServer - Listening for HTTP on /127.0.0.1:9000

(Server started, use Enter to stop and go back to the console...)

[info] p.a.i.l.c.CoordinatedShutdownSupport - Starting synchronous coordinated shutdown with ServerStoppedReason reason and 35000 milliseconds timeout
[info] p.c.s.AkkaHttpServer - Stopping Akka HTTP server...
[info] p.c.s.AkkaHttpServer - Terminating server binding for /127.0.0.1:9000
[info] p.c.s.AkkaHttpServer - Running provided shutdown stop hooks

[success] Total time: 4 s, completed Jan 6, 2022, 10:07:43 PM
```

**Comment by mkurz** [comment]
Thanks for the report, I will have a look the next days.
Fortunately, meanwhile you can still run `sbt run` from the command line.
@gmethvin I am pretty sure this a is a consequence of #10890.

**Comment by gmethvin** [comment]
It seems like this might be a Windows-specific bug, as I'm not able to reproduce on macOS.

I created an equivalent shell script:
```sh
#!/bin/sh
sbt run > console.log 2>&1
```
and that works as expected.

So I wonder what Windows does that's different here...

**Comment by mkurz** [comment]
I can reproduce this, let's see if we can fix this on Play's side or if IntelliJ  has to do something against it...

**Comment by mkurz** [comment]
@greg Did you use sbt 1.4 or newer? Because I can reproduce on my Ubuntu machine with 1.4.9, 1.5.8 and 1.6.1, but not with 1.3.13 and below. Also I think I have a clue already what's going on...

**Comment by gmethvin** [comment]
On macOS it works as expected for me on sbt 1.6.1 and Play 2.8.12.

**Comment by mkurz** [comment]
@MikeEngUSA @gmethvin Here is a fix: #11088 Can you please help to test it? Instructions in the PR comment, thanks! If everyhing goes well, I will release 2.8.13.

**Comment by MikeEngUSA** [comment]
@mkurz: Thanks for the rapid fix! I'm stuck in meetings this morning, but I'll test it this afternoon and report back whether it works for me.

**Comment by mkurz** [comment]
@MikeEngUSA No hurry! Thanks!

**Comment by MikeEngUSA** [comment]
@mkurz I ran the tests and it worked for me. Further comments are logged against fix #11088. Thanks!

**Comment by mkurz** [comment]
Fixed by #11088

**Comment by EirikFinvold** [comment]
I face what seems like the same issue with 2.8.15 and the latest IntelliJ Idea (IntelliJ IDEA 2022.1 (Ultimate Edition) Build #IU-221.5080.210, built on April 12, 2022). Just like with the problem above it works fine running `sbt run` but not Idea->Run->Debug.
It works perfectly well on linux, but not on two different Windows computers.

Reproduce: 
- Create a brand new project with sbt new playframework/play-java-seed.g8
- Open with Idea
- Use run->debug->sbt task 
IntelliJ is configured with Oracle OpenJDK 11.0.14, SBT Shell use for both "project reloads" and "build".
"Use Play 2 compiler for this project" is disabled (tried both).

When starting the debugger it builds just fine, but when it after building opens the debugger console it exits immediately:

`Connected to the target VM, address: '127.0.0.1:52457', transport: 'socket'`
`Disconnected from the target VM, address: '127.0.0.1:52457', transport: 'socket'`
` `
`Process finished with exit code 2`

Closing Idea and running `call sbt run > test.log 2>&1` works, but maybe the first error is the problem?

_A subdirectory or file [info] welcome to sbt 1.6.2 (Oracle Corporation Java 11.0.14) already exists._
_[info] welcome to sbt 1.6.2 (Oracle Corporation Java 11.0.14)_
_[info] loading settings for project global-plugins from idea.sbt ..._
_[info] loading global plugins from C:\Users\Eirik\.sbt\1.0\plugins_
_[info] loading settings for project microlog-learn-play-build from plugins.sbt ..._
_[info] loading project definition from C:\Users\Eirik\microlog-learn-play\project_
_[info] loading settings for project root from build.sbt ..._
_[info] set current project to microlog-learn-play (in build file:/C:/Users/Eirik/microlog-learn-play/)_
_[error] Expected 'error'_
_[error] --export-rt_
_[error]    ^_
_[info] welcome to sbt 1.6.2 (Oracle Corporation Java 11.0.14)_
_[info] loading settings for project global-plugins from idea.sbt ..._
_[info] loading global plugins from C:\Users\Eirik\.sbt\1.0\plugins_
_[info] loading settings for project microlog-learn-play-build from plugins.sbt ..._
_[info] loading project definition from C:\Users\Eirik\microlog-learn-play\project_
_[info] loading settings for project root from build.sbt ..._
_[info] set current project to microlog-learn-play (in build file:/C:/Users/Eirik/microlog-learn-play/)_

_--- (Running the application, auto-reloading is enabled) ---_

_[info] p.c.s.AkkaHttpServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000_

_(Server started, use Enter to stop and go back to the console...)_


I tried SBT versions from 1.4.0->1.6.2 and also downgrading Play back to 2.8.8.

**Comment by mkurz** [comment]
Thanks for the report. I don't have access to IntelliJ on a Windows machine right now, so it's a bit hard for me to reproduce that issue. Since you can confirm this problem on 2 independent machines running Windows, can you please open a new dedicated issue since this one here already is closed and got the 2.8.13 milestone attached on it. Thanks!

**Comment by EirikFinvold** [comment]
Sure, thank you for your quick reply! Have a nice weekend.
